### PR TITLE
feat(interactive): Support `PathExpand` with `PATH_OPT=TRAIL` in GIE

### PR DESCRIPTION
**Pull request by fangfcg**
## What do these changes do?
This PR introduces a new option `TRAIL` for Path Expand by specifying `PATH_OPT` as`TRAIL`. It ensures no duplicated edges (i.e., unique edges) during the traversals, e.g., 
 
example

### Configurations
Currently, we support `PATH_OPT` values `ARBITRARY`, `SIMPLE` and `TRAIL`. The differences are as follows:

 Parameter | Supported Values  | Description                                    
-----------|-------------------|------------------------------------------------
 PATH_OPT  | ARBITRARY         | Allow vertices or edges to be duplicated.     
 PATH_OPT  | SIMPLE            | No duplicated nodes.            
 PATH_OPT  | TRAIL             | No duplicated edges.            
 RESULT_OPT| END_V             | Only keep the end vertex.                      
 RESULT_OPT| ALL_V             | Keep all vertices along the path.              
 RESULT_OPT| ALL_V_E           | Keep all vertices and edges along the path.    

### Example
from A to X:
- **Arbitrary path** (`PATH_OPT=ARBITRARY`): Infinite paths in the figure.
- **Simple path** (`PATH_OPT=SIMPLE`): Just one path A->B->C->D->X
- **Trail** (unique edge, `PATH_OPT=TRAIL`): Two paths, A->B->C->D->X, and A->B->C->E->F->C->D->X

image

The term "trail" is defined in [1].

## Related issue number

Fixes [#4072](https://github.com/alibaba/GraphScope/issues/4072)

[1] Deutsch, Alin, et al. "Graph pattern matching in GQL and SQL/PGQ." Proceedings of the 2022 International Conference on Management of Data. 2022.

### CODE REVIEW

File: interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs [hunk 1]
```diff
@@ -149,41 +168,55 @@ impl GraphPath {
                     true
                 }
             }
+            GraphPath::TrailEndV((ref mut e, ref mut path, ref mut weight)) => {
+                let entry = entry.into();
+                if entry.is_edge() && path.contains(&entry.id()) {
```
[the remark this answers is not in the document]
Comment (fangfcg) [reply interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs]: Add support for `Trail` with `ALL_V` and `END_V` in the new commit.

File: interactive_engine/executor/ir/runtime/src/assembly.rs [hunk 2]
```diff
@@ -786,7 +786,10 @@ impl<P: PartitionInfo, C: ClusterInfo> IRJobAssembly<P, C> {
                             base
                         )))
                     })?;
-
+                    if pb::path_expand::PathOpt::Trail == unsafe { std::mem::transmute(path.path_opt) }
```
[the remark this answers is not in the document]
Comment (fangfcg) [reply interactive_engine/executor/ir/runtime/src/assembly.rs]: Add support for `Trail` with `ALL_V` and `END_V` in the latest commit.

File: interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs [hunk 3]
```diff
@@ -127,6 +137,15 @@ impl GraphPath {
                     true
                 }
             }
+            GraphPath::TrailAllPath(ref mut path) => {
+                let entry = entry.into();
+                if entry.is_edge() && path.contains(&entry) {
```
[the remark this answers is not in the document]
Comment (fangfcg) [reply interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs:141]: Added.

File: interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs [hunk 4]
```diff
@@ -149,41 +167,55 @@ impl GraphPath {
                     true
                 }
             }
+            GraphPath::TrailEndV((ref mut e, ref mut path, ref mut weight)) => {
```
[the remark this answers is not in the document]
Comment (fangfcg) [reply interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs]: revised

**Review by BingqingLyu — DISMISSED** [review]
LGTM.

**Review by shirly121 — DISMISSED** [review]

**Review by BingqingLyu — APPROVED** [review]

**Review by BingqingLyu — APPROVED** [review]

**Review by shirly121 — DISMISSED** [review]

**Review by shirly121 — APPROVED** [review]

**Review by BingqingLyu — APPROVED** [review]

**Pull request merged**